### PR TITLE
Added Trie [CPP]

### LINKED ISSUE #334: Trie [CPP]
#### This issue is about [Trie](https://en.wikipedia.org/wiki/Trie).
- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #334

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: trie/trie.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		void insert(string key);
+
+		// Search into the Trie, Implementation is below
+		bool search(string key);
+};
+
+// Insert into the Trie
+// args :
+//    key : String to insert into Trie
+// Time Complexity: O(len(key))
```
Comment (aviaryan) [inline trie/trie.cpp]: Use `/* ... */` blocks for function comments. See https://stackoverflow.com/questions/51667/best-tips-for-documenting-code-using-doxygen

File: trie/trie.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+// TrieNode class
+class TrieNode {
+	private:
+		// Children of every Node
+		class TrieNode *children[ALPHABET_SIZE];
+		// Status of Leaf Node
+		bool isLeaf;
+	public:
+
```
Comment (aviaryan) [inline trie/trie.cpp]: Empty line not needed here.. rather there should be an empty line after `private:` scope and before starting of `public`.

**Comment by aviaryan** [comment]
@yashLadha Nice work. Please resolve the issues.

File: trie/trie.cpp (hunk 2 above)
Comment (yashLadha) [inline trie/trie.cpp]: Done :+1:

File: trie/trie.cpp (hunk 1 above)
Comment (yashLadha) [inline trie/trie.cpp]: Implemented Sir :+1:

File: trie/trie.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
+#include <iostream>
+#include <cstring>
+using namespace std;
+
+// Determine size of array
+#define ARRAY_SIZE(a) sizeof(a)/sizeof(a[0])
```
Comment (Monal5031) [inline trie/trie.cpp]: Space around binary operator here.

Comment (yashLadha) [inline trie/trie.cpp]: Done :+1:

File: trie/trie.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+int main() {
+	// keys to insert into Trie
+	char keys[][8] = {"the", "a", "there", "answer", "any",
+		"by", "bye", "their"};
+	// Output the status of key
+	char output[][32] = {"Not present in trie", "Present in trie"};
+	// Root of Trie (Crawling starts from root, insertion as well)
+	class TrieNode *root = new TrieNode();
+	// Insertion into Trie
+	for (unsigned long i = 0; i < ARRAY_SIZE(keys); ++i)
```
Comment (aviaryan) [inline trie/trie.cpp]: why use `unsigned long` here?

**Comment by yashLadha** [comment]
Sir the syntastic flagged error to use an unsigned long value. Syntastic is
code linter and syntax checker for vim.

**Pull request merged**